src/test/java/com/google/jenkins/plugins/computeengine/ConfigAsCodeTest.java

```diff
@@ -51,4 +52,16 @@ public void shouldCreateGCEClientFromCode() throws Exception {
         // Ensure correct exception is thrown
         assertThrows(GoogleRobotPrivateKeyCredentials.PrivateKeyNotSetException.class, cloud::getClient);
     }
+
+    @Test
+    @ConfiguredWithCode("casc-preemptible-compatibility.yml")
+    public void provisioningTypeShouldBePreemptible() {
+        ComputeEngineCloud cloud = (ComputeEngineCloud) jenkinsRule.jenkins.clouds.getByName("gce-jenkins-build");
+        assertEquals("Zero configurations found", 1, cloud.getConfigurations().size());
+        InstanceConfiguration configuration = cloud.getConfigurations().get(0);
+        assertEquals(
+                "Provisioning type is wrong",
+                ProvisioningTypeValue.PREEMPTIBLE,
+                configuration.getProvisioningType().getValue());
```

Comment:
Assuming you get rid of the enum (or not)

```suggestion
        assertThat(cloud.getConfigurations(), contains(instanceOf(PreemptibleVm.class)));
```